apps/cowswap-frontend/src/modules/tokensList/pure/SelectTokenModal/index.tsx

```diff
@@ -105,64 +32,191 @@ export function SelectTokenModal(props: SelectTokenModalProps): ReactNode {
+      onDismiss={onDismiss}
+      onOpenManageWidget={onOpenManageWidget}
+      searchValue={inputValue}
+      onSearchChange={setInputValue}
+      onSearchEnter={onInputPressEnter}
+    >
+      <TokenColumnContent
+        displayLpTokenLists={displayLpTokenLists}
+        account={account}
+        inputValue={inputValue}
+        onSelectToken={onSelectToken}
+        openPoolPage={openPoolPage}
+        disableErc20={disableErc20}
+        tokenListCategoryState={tokenListCategoryState}
+        isRouteAvailable={isRouteAvailable}
+        chainsToSelect={legacyChainsState}
+        onSelectChain={onSelectChain}
+      >
+        <TokensContentSection
+          displayLpTokenLists={displayLpTokenLists}
+          favoriteTokens={favoriteTokens}
+          recentTokens={recentTokens}
+          onClearRecentTokens={onClearRecentTokens}
+          onOpenManageWidget={onOpenManageWidget}
+          standalone={standalone}
+          areTokensLoading={areTokensLoading}
+          allTokens={allTokens}
+          searchInput={trimmedInputValue}
+          areTokensFromBridge={areTokensFromBridge}
+          hideFavoriteTokensTooltip={hideFavoriteTokensTooltip}
+          selectedTargetChainId={selectedTargetChainId}
+          selectTokenContext={selectTokenContext}
+        />
+      </TokenColumnContent>
+    </SelectTokenModalShell>
+  )
+}
 
-  const allListsContent = (
-    <TokensContent
-      {...props}
-      selectTokenContext={selectTokenContext}
-      searchInput={trimmedInputValue}
-      areTokensFromBridge={areTokensFromBridge}
-    />
+interface TokenColumnContentProps {
+  displayLpTokenLists?: boolean
+  account: string | undefined
+  inputValue: string
+  onSelectToken: TokenSelectionHandler
+  openPoolPage(poolAddress: string): void
+  disableErc20?: boolean
+  tokenListCategoryState: SelectTokenModalProps['tokenListCategoryState']
+  isRouteAvailable: boolean | undefined
+  chainsToSelect?: SelectTokenModalProps['chainsToSelect']
+  onSelectChain: SelectTokenModalProps['onSelectChain']
+  children: ReactNode
+}
+
+function TokenColumnContent({
+  displayLpTokenLists,
+  account,
+  inputValue,
+  onSelectToken,
+  openPoolPage,
+  disableErc20,
+  tokenListCategoryState,
+  isRouteAvailable,
+  chainsToSelect,
+  onSelectChain,
+  children,
+}: TokenColumnContentProps): ReactNode {
+  if (displayLpTokenLists) {
+    return (
+      <LpTokenListsWidget
+        account={account}
+        search={inputValue}
+        onSelectToken={onSelectToken}
+        openPoolPage={openPoolPage}
+        disableErc20={disableErc20}
+        tokenListCategoryState={tokenListCategoryState}
+      >
+        {children}
+      </LpTokenListsWidget>
+    )
+  }
+
+  return (
+    <>
+      {renderLegacyChainSelector(chainsToSelect, onSelectChain)}
+      <SelectTokenModalContent isRouteAvailable={isRouteAvailable}>{children}</SelectTokenModalContent>
+    </>
   )
+}
+
+function renderLegacyChainSelector(
```

Comment:
also will be better to move it from this file

Reply:
Thank you. Addressed in https://github.com/cowprotocol/cowswap/commit/b3f48f1525aeb39959a7f8f634f290e8dc9b4c1e